File: assignment2/1.2/section1.2.js
// JavaScript: Scopes and Closures
function outerFunction() {
    let x = 10;  // Variable in the outer function scope
    
    function innerFunction() {
        console.log(x);  // Inner function uses variable from outer scope (closure)
    }
    
    innerFunction();
}

outerFunction();  // Output: 10

File: assignment2/1.2/section1.2.cpp
// C++: Scopes and Closures
#include <iostream>
using namespace std;

int main() {
    int x = 10;  // Variable in the local scope

    auto closure = [x]() {  // Lambda function captures x from the surrounding scope
        cout << x << endl;
    };

    closure();  // Output: 10
    return 0;
}
